Model/Api/ShippingMethods.php

```diff
@@ -526,15 +535,22 @@ private function getShippingOptions($quote, $addressData)
             ////////////////////////////////////////////////////////////////
 
             $shippingAddress->setShippingMethod($method);
+            // in order to get correct shipping discounts
+            // the following method must be called twice
+            $this->totalsCollector->collectAddressTotals($quote, $shippingAddress);
```

Comment:
I'm not sure how similar this is to running collectTotals() on Magento 1, but if they are the same, this workaround may cause unintended consequences later on since it's based on calling collectTotals an even number of times (the second time rounds it back down). If so, you may want to look at what was done here https://github.com/BoltApp/bolt-magento1/commit/59fa6c3f377c62b7c51a149d4edbb3710b49789c